### PR TITLE
[ESIMD] Add smoke test for pack_mask/unpack_mask APIs.

### PR DESCRIPTION
**Pull request by kbobrovs**
Complementary E2E test for https://github.com/intel/llvm/pull/5547

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+
+  bool passed = true;
+  passed &= test<1>(q);
+  passed &= test<2>(q);
+  passed &= test<7>(q);
+  passed &= test<8>(q);
+  passed &= test<16>(q);
+  // TODO disabled due to compiler bug
```
Comment (kbobrovs) [inline SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp:126]:
```suggestion
  // TODO disabled due to a compiler bug
```

File: SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+
+  bool passed = true;
+  passed &= test<1>(q);
+  passed &= test<2>(q);
+  passed &= test<7>(q);
+  passed &= test<8>(q);
+  passed &= test<16>(q);
+  // TODO disabled due to a compiler bug
```
Comment (kbobrovs) [inline SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp]:
```suggestion
  // TODO disabled due to compiler bug
```

File: SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// TODO: esimd_emulator fails due to unimplemented 'single_task()' method
+// XFAIL: esimd_emulator
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Smoke test for esimd pack_mask/unpack_mask APIs.
```
Comment (kbobrovs) [inline SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp]:
```suggestion
// Smoke test for the esimd pack_mask/unpack_mask APIs.
```

File: SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    auto e = q.submit([&](handler &cgh) {
+      cgh.single_task<test_id<N, TestCase>>([=]() SYCL_ESIMD_KERNEL {
+        simd_mask<N> m(test_data);
+        uint32_t packed_m = pack_mask(m);
+        res_packed[0] = packed_m;
+        unpack_mask<MAX_N>(packed_m).copy_to(res_unpacked);
+      });
+    });
+    e.wait_and_throw();
+  } catch (cl::sycl::exception const &e) {
```
Comment (v-klochkov) [inline SYCL/ESIMD/api/esimd_pack_unpack_mask.cpp]:
```suggestion
  } catch (sycl::exception const &e) {
```

**Review by v-klochkov — DISMISSED** [review]
Looks good. I have only one non-blocking comment

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**